graphql_compiler/tests/test_ir_generation_errors.py

```diff
@@ -196,16 +196,28 @@ def test_fold_directive_constraints(self):
             }
         }'''
 
-        traversal_inside_fold_block = '''{
+        multi_level_outputs_inside_fold_block = '''{
             Animal {
                 out_Animal_ParentOf @fold {
+                    uuid @output(out_name: "uuid")
                     out_Animal_FedAt {
                         uuid @output(out_name: "uuid")
                     }
                 }
             }
         }'''
 
+        traversal_inside_fold_block_after_output = '''{
+            Animal {
+                out_Animal_ParentOf @fold {
+                    uuid @output(out_name: "uuid")
+                    out_Animal_FedAt {
+                        uuid
+                    }
+                }
+            }
+        }'''
```

Comment:
can you also add a "tree-like" traversal attempt, where all outputs are in the innermost scopes but there are multiple such scopes in the `@fold`?

Reply:
Done!